actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -675,6 +675,18 @@ def app=(app)
           @@app = app
         end
 
+        # Registers an encoder to encode request parameters and to parse
+        # response body for a MIME type.
+        #
+        # #### Example
+        #
+        #     ActionDispatch::IntegrationTest.register_encoder :xml,
+        #       param_encoder: -> params { params.to_xml },
+        #       response_parser: -> body { Hash.from_xml(body) }
+        #
+        # `param_encoder` encodes request parameters when performing a request.
+        # `response_parser` parses response body through
+        # `ActionDispatch::TestResponse#parsed_body`.
```

Comment:
Does this differ substantially from https://github.com/rails/rails/pull/49513 (particularly [this block](https://github.com/rails/rails/pull/49513/files#diff-7c57a19207aafe25adcf81324c922c9bda0205db771c91f5365de6eb10197b53R676-R688))?

Reply:
Oh, I didn't realize that PR. I'm going to look at it and leave comments if any. Thank you!